### PR TITLE
Create yield.rst:: Addresses issue #2546 "DOC Add explanation of retu…

### PR DESCRIPTION
**Pull request by neonwuchang**
…rn vs. yield in callbacks"

Addresses issue #2546 "DOC Add (short) explanation of return vs. yield in callbacks" 

This commit adds a file 'yield.rst' which contains a basic explanation of 'yield' keyword in Python, how it works and is different from 'return', and when to use it, with some simple sample code, to docs/topics.

### CODE REVIEW

**Comment by wRAR** [comment]
I'm afraid none of this talks about Scrapy callbacks.

**Comment by neonwuchang** [comment]
Looks like I misunderstood that, my bad.

**Comment by neonwuchang** [comment]
I'll make the changes

**Comment by neonwuchang** [comment]
@wRAR could you please take a look to see if this in more in line with what's needed?

Thanks

File: docs/topics/yield.rst [hunk 1]
```diff
@@ -0,0 +1,176 @@
+.. _topics-yield:
+
+==========================
+'Yield' Keyword in Python and its use for Scrapy Callbacks
+==========================
+  
+What is ``yield``?
```
Comment (wRAR) [inline docs/topics/yield.rst:7]: I don't think we need such a long explanation what are Python generator and how does the yield keyword in Python work, there are better existing places for that.

File: docs/topics/yield.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    return 5
+  res = gen()
+  for x in res:
+    print(x)
+
+.. _topics-yield-scrapy:
+
+Where is ``yield`` used in Scrapy?
+==================================
+``Yield`` is used for callbacks in Scrapy, which help scrape multiple pages. For example, consider the following code
```
Comment (wRAR) [inline docs/topics/yield.rst:121]: I don't think the "which help scrape multiple pages" part makes sense here?

**Comment by wRAR** [comment]
I'm also not sure if this should be a separate page, as these two paragraphs are better placed next to some related docs.

**Comment by neonwuchang** [comment]
I absolutely see and agree with the point that there are better places for detailed explanations of generators and yield keyword outside the Scapy docs. (I was being extra thorough given that this was meant to help beginners). 

In this case, having a separate page is certainly not needed.

**Comment by neonwuchang** [comment]
Moving forward with the assumption that the explanations are removed, would it be right to say that the content under the last two paragraphs ("Why use yield?" and "Would return work?"), is the kind of content that addresses the issue, or is that off the mark? It would be helpful to have some direction

Thanks!

**Comment by neonwuchang** [comment]
I'll close this PR as I won't be able to get to this for a while because of irl tasks and I don't want to leave this hanging. And if the issue is still open, I can work on it at a later date.

**Pull request closed without merging**